src/main/kotlin/org/rust/lang/core/resolve/Processors.kt

```diff
@@ -81,6 +81,22 @@ fun collectResolveVariants(referenceName: String, f: (RsResolveProcessor) -> Uni
     return result
 }
 
+fun <T : ScopeEntry> collectResolveVariantsAsSEs(referenceName: String, f: ((T) -> Boolean) -> Unit): List<T> {
+    val result = mutableListOf<T>()
+    f { e ->
+        if ((e == ScopeEvent.STAR_IMPORTS) && result.isNotEmpty()) {
+            return@f true
+        }
+
+        if (e.name == referenceName) {
+            e.element ?: return@f false // de-lazying
```

Comment:
Let's add a link to `org.rust.lang.core.resolve.Processors#lazy` method and add a note to method's doc about it because it looks quite important